internal/rbd/rbd_attach.go

```diff
@@ -48,10 +48,22 @@ const (
 	rbdUnmapCmdNbdMissingMap  = "rbd-nbd: %s is not mapped"
 	rbdMapConnectionTimeout   = "Connection timed out"
 
-	defaultNbdReAttachTimeout = 300
+	defaultNbdReAttachTimeout = 300 /* in seconds */
+	defaultNbdIOTimeout       = 0   /* do not abort the requests */
 
-	useNbdNetlink  = "try-netlink"
+	// The default way of creating nbd devices via rbd-nbd is though the
+	// legacy iotcl interface, to take advantage of netlink features we
+	// should specify `try-netlink` flag explicitly.
+	useNbdNetlink = "try-netlink"
+
+	// `reattach-timeout` of rbd-nbd is to tweaks NBD_ATTR_DEAD_CONN_TIMEOUT.
+	// It specifies how long the device should be hold waiting for the
+	// userspace process to come back to life.
 	setNbdReattach = "reattach-timeout"
+
+	// `io-timeout` of rbd-nbd is to tweak NBD_ATTR_TIMEOUT. It specifies
+	// how long the IO should wait to get handled before bailing out.
```

Comment:
it would be better if we can also add some reference to the mentoned timeout attributes from ceph core doc urls.

Reply:
s/mentoned/mention/

There are no reference docs AFAIK.